components/lib/jexl/src/main/java/mozilla/components/lib/jexl/value/JexlValue.kt

```diff
@@ -0,0 +1,304 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.jexl.value
+
+import mozilla.components.lib.jexl.evaluator.EvaluatorException
+
+/**
+ * A JEXL value type.
+ */
+sealed class JexlValue {
+    abstract operator fun plus(other: JexlValue): JexlValue
+    abstract operator fun times(other: JexlValue): JexlValue
+    abstract operator fun div(other: JexlValue): JexlValue
+    abstract operator fun compareTo(other: JexlValue): Int
+
+    abstract fun toBoolean(): Boolean
+    abstract fun toKotlin(): Any
```

Comment:
I found `toKotlin` a bit confusing. Maybe just `value` (to map the constructor param)?